configs/installerconfig

```diff
@@ -0,0 +1,207 @@
+        test -n "$DISTDIR_IS_UNIONFS" && umount -f $BSDINSTALL_DISTDIR
+        test -f $PATH_FSTAB && bsdinstall umount
+        dialog --backtitle "FreeBSD Installer" --title "Abort" \
+            --no-label "Exit" --yes-label "Restart" --yesno \
+            "${msg}An installation step has been aborted. Would you like to rest
+art the installation or exit the installer?" 0 0
+        if [ $? -ne 0 ]; then
+        exit 1
+else
+exec $0
+fi
+}
+
+############################################################ MAIN
+
+f_dprintf "Began Installation at %s" "$( date )"
+
+rm -rf $BSDINSTALL_TMPETC
+mkdir $BSDINSTALL_TMPETC
+
+export DISTRIBUTIONS="base.txz kernel.txz"
+if [ -f $BSDINSTALL_DISTDIR/MANIFEST ]; then
+DISTMENU=`awk -F'\t' '!/^(kernel|base)/{print $4,$5,$6}' $BSDINSTALL_DIS
+TDIR/MANIFEST`
+fi
+
+FETCH_DISTRIBUTIONS=""
+for dist in $DISTRIBUTIONS; do
+        if [ ! -f $BSDINSTALL_DISTDIR/$dist ]; then
+                FETCH_DISTRIBUTIONS="$FETCH_DISTRIBUTIONS $dist"
+        fi
+done
+FETCH_DISTRIBUTIONS=`echo $FETCH_DISTRIBUTIONS` # Trim white space
+
+echo ifconfig_$INTERFACE=\"DHCP\">> $BSDINSTALL_TMPETC/._rc.conf.net
+err=$( dhclient $INTERFACE 2>&1 )
+if [ $? -ne 0 ]; then
+        f_dprintf "%s" "$err"
+fi
+
+
+IP_ADDRESS=`ifconfig $INTERFACE inet | awk '/inet/ {printf("%s\n", $2); }'`
+NETMASK=`ifconfig $INTERFACE inet | awk '/inet/ {printf("%s\n", $4); }'`
+ROUTER=`netstat -rn -f inet | awk '/default/ {printf("%s\n", $2);}'`
+
+SEARCH=""
+IP4_1=""
+IP4_2=""
+IP6_1=""
+IP6_2=""
+
+RESOLV="
+            'Search' 1 0 \"${SEARCH}\" 1 16 50 0 0
+            'Nameserver' 2 0 \"Nameserver\" 2 16 50 0 2
+            'IPv4 DNS #1' 2 0 \"${IP4_1}\" 2 16 16 0 0
+            'IPv4 DNS #2' 3 0 \"${IP4_2}\" 3 16 16 0 0"
+
+echo ${RESOLV} | tr ' ' '\n' | \
+awk '
+BEGIN {
+        search=-1;
+}
+{
+        if (/^[[:space:]]+$/) {
+                next;
+        }
+        if (/^Nameserver$/) {
+        printf "\n";
+        search=0;
+        next;
+}
+if (search == -1) {
+        printf "search ";
+        search=1;
+}
+if (search > 0) {
+        printf "%s%s", (search > 1) ? " " : "", $1;
+        search++;
+        next;
+}
+printf "nameserver %s\n", $1;
+}' > ${BSDINSTALL_TMPETC}/resolv.conf
+
+mv $BSDINSTALL_TMPETC/._rc.conf.net $BSDINSTALL_TMPETC/rc.conf.net
+
+NETCONFIG_DONE=yes
+
+if [ -n "$FETCH_DISTRIBUTIONS" ]; then
+    _UNAME_R=`uname -r`
+    _UNAME_R=${_UNAME_R%-p*}
+    case ${_UNAME_R} in
+        *-CURRENT|*-STABLE|*-PRERELEASE)
+                RELDIR="snapshots"
+                ;;
+        *)
+                RELDIR="releases"
+                ;;
+    esac
+    BSDINSTALL_DISTSITE="$MIRROR/pub/FreeBSD/${RELDIR}/`uname -m`/`uname -p`/${_UNAME_R}"
+    export BSDINSTALL_DISTSITE
+fi
+
+
+rm -f $PATH_FSTAB
+touch $PATH_FSTAB
+
+bsdinstall zfsboot || error "ZFS setup failed"
+bsdinstall mount || error "Failed to mount filesystem"
+
+if [ ! -z "$FETCH_DISTRIBUTIONS" ]; then
+        ALL_DISTRIBUTIONS="$DISTRIBUTIONS"
+
```

Comment:
I do not quite understand why we have to fetch a kernel from mirror. Isn't it available in the installer image already? @adamwawrzynski3mdeb any ideas?

Reply:
Fetching kernel is excess, but in original (manual) USB-stick installation MANIFEST and base.txz file is fetched.